### DISCUSSION TITLE
Need of --host-resolver-rules flag when configuring a proxy

### DISCUSSION THREAD

**Discussion by miquecg** [bromite/bromite#2401] [Q&A]
Hi,

I've moved away from a `chrome-command-line` file to configure a proxy on bromite. I'm now just using the available settings for it.

Previously I followed the advice from [here](https://www.chromium.org/developers/design-documents/network-stack/socks-proxy/), which suggests to also block DNS resolution for other components.

> The --proxy-server flag applies to URL loads only. There are other components of Chrome which may issue DNS resolves directly and hence bypass this proxy server. The most notable such component is the "DNS prefetcher". Hence if DNS prefetching is not disabled in Chrome then you will still see local DNS requests being issued by Chrome despite having specified a SOCKS v5 proxy server.

Does it still apply when configuring proxies from application settings? If that's the case, maybe I can run both command line flags and a proxy configured in settings.

> _ --host-resolver-rules="MAP * ~NOTFOUND , EXCLUDE *.proxy.com"

Proxy would be `something.proxy.com` directly configured on the settings.

**Comment by uazo** [comment]
I would say that it might be a good idea to disable prefetch or to run it through the proxy, if possible.
have you checked whether this actually happens? would you be able to prepare a test page?

this seems to be the class [preload_helper.cc](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/preload_helper.cc)
there is also a flag on the blink settings [GetDNSPrefetchingEnabled](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/preload_helper.cc;l=173;drc=4be70d0a54a18a36221e3733fd1dea827fc5f7ad;bpv=1;bpt=1) default [true](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/common/web_preferences/web_preferences.cc;l=58;drc=4be70d0a54a18a36221e3733fd1dea827fc5f7ad;bpv=1;bpt=1?q=WebPreferences::WebPreferences)

**Reply by miquecg** [reply]
For now I'm fine disabling prefetch on settings.

> https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/preload_helper.cc;l=186

Does this mean that DNS queries in this module bypass system configured resolver? If that's the case, I think it would be great for Bromite to have a way for avoiding it, at least when the user proactively sets a proxy.

**Reply by csagan5** [reply]
I think the proxy code needs a non-proxied DNS resolver to resolve names present in the proxy rules; only those requests should happen outside of the proxy; then there is also to consider the extra complexity related to DoH vs system DNS resolver.
